libr/bin/format/elf/elf.c

```diff
@@ -1014,6 +1014,11 @@ ut64 Elf_(r_bin_elf_get_baddr)(struct Elf_(r_bin_elf_obj_t) *bin) {
 			}
 		}
 	}
+	if (base == UT64_MAX  && bin->ehdr.e_type == ET_REL) {
+		//we return our own base address for ET_REL type
+		//we act as a loader for ELF
+		return 0x08000000;
```

Comment:
i choose that addr to avoid overlap since the vaddr before was 0 for all sections, now is that address+paddr. Bear in mind that you are dealing with a relocatable file so there isn't a phdr. So I choose my own base address, to load the different sections there (iirc IDA also loads the binary into 0x08000000). If you fire up IDA or Hopper each one produces different output, so anyway you can't share that information. 

Regarding the relocation i create my own section `.got.r2` to point all the relocation there and patch the binary. So in master now you see something like

```
;-- check_free_space:
            0x00000080      e800000000     call 0x85
```

with my patch

```
 ;-- check_free_space:
            0x08000080      e8c72d0000     call reloc.__fentry___76    ;[5]
```